docs/user/openstack/ovs-dpdk.md

```diff
@@ -0,0 +1,244 @@
+The performance-operator is used to configure these settings on the desired worker nodes.
+For details on installing and using the performance-operator see the [official documentation][2].
+
+The following is an example of the performance profile that should be applied.
+The user should provide the values for kernel args, `HUGEPAGES`, `CPU_ISOLATED` and `CPU_RESERVED` depending on their system:
+
+``` yaml
+apiVersion: performance.openshift.io/v1
+kind: PerformanceProfile
+metadata:
+  name: cnf-performanceprofile
+spec:
+  additionalKernelArgs:
+    - nmi_watchdog=0
+    - audit=0
+    - mce=off
+    - processor.max_cstate=1
+    - idle=poll
+    - intel_idle.max_cstate=0
+    - default_hugepagesz=1GB
+    - hugepagesz=1G
+    - intel_iommu=on
+  cpu:
+    isolated: <CPU_ISOLATED>
+    reserved: <CPU_RESERVED>
+  hugepages:
+    defaultHugepagesSize: 1G
+    pages:
+      - count: <HUGEPAGES>
+        node: 0
+        size: 1G
+  nodeSelector:
+    node-role.kubernetes.io/worker: ''
+  realTimeKernel:
+    enabled: false
+```
+
+### Enable `vfio-noiommu` driver
+
+In the case of deploying workers in virtual machines, the underlying virtualization platform doesn't support
+a virtualized IOMMU and we need to explicitly enable it via `options vfio enable_unsafe_noiommu_mode=1`.
+This can be done via a MachineConfig resource:
+
+``` yaml
+kind: MachineConfig
+apiVersion: machineconfiguration.openshift.io/v1
+metadata:
+  name: 99-vfio-noiommu 
+  labels:
+    machineconfiguration.openshift.io/role: worker
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+      - path: /etc/modprobe.d/vfio-noiommu.conf
+        mode: 0644
+        contents:
+          source: data:;base64,b3B0aW9ucyB2ZmlvIGVuYWJsZV91bnNhZmVfbm9pb21tdV9tb2RlPTEK
+```
+
+More documentation about that can be found [here][3].
+
+### Bind the `vfio-pci` kernel driver to the appropriate NICs
+
+The workers connected to the `VFIO_NETWORK` require the `vfi-pci` kernel driver to be bound to the ports attached to the `VFIO_NETWORK`.
+
+We basically need to create a MachineSet for the workers which will be attached to the `VFIO_NETWORK` network.
+The MachineConfig used in the MachineSet should contain the necessary SystemD unit definitions to install the poll mode driver (`vfio-pci`) on boot.
+This [Ansible role][4] provides an easy way to generate the required MachineSet.
+The playbooks also provide the ability to automatically install the MachineSet, but it is highly recommended to use the playbooks for generating a MachineSet file (.yaml).
+Then the user should explicitly install this on their cluster using the appropriate `oc` commands.
+
+To generate the MachineSet file, run the following commands:
+
+``` bash
+openstack network show <name of VFIO network> -f value -c id
+ansible-playbook play.yaml -e network_ids="ID of VFIO network" --extra-vars "mc_config_file=/tmp/vhostuser.yaml"
```

Comment:
In the case of multiple networks, the comma-separated IDs can be provided. should this be added here?